### ISSUE TITLE
[Bug]: Running page.screenshot in parallel is blocking all calls

### ISSUE THREAD

**Issue by GuillaumeGomez** [puppeteer/puppeteer#12670]
### Minimal, reproducible example

```TypeScript
Calling `page.screenshot` from multiple tests at the same time.
```

### Error string

no error

### Bug behavior

- [ ] Flaky
- [ ] PDF

### Background

I run a lot of GUI tests in parallel and some of them are generating a screenshot at the end. I upgraded puppeteer from version 19.4.0 to 22.12.0 and this bug appeared. Now, all these tests are blocking as soon as I call `page.screenshot`. After investigating, it only occurs when multiple tests are doing at the same time.

Another note: if it's not a full page screenshot, it'll fail after the timeout expires.

### Expectation

To work.

### Reality

It doesn't.

### Puppeteer version

22.12.1

### Node version

v20.15.0

### Package manager

npm

### Package manager version

10.7.0

### Operating system

Linux

**Comment by GuillaumeGomez** [comment]
Seems to be fixed in 22.12.1. Leaving it open just in case so regression tests for this issue are added. :wink:

**Comment by OrKoN** [comment]
I am unable to reproduce and we do have a test for this scenario https://github.com/puppeteer/puppeteer/blob/18863902747f84841a46e39673f6ea69a23fddd1/test/src/screenshot.spec.ts#L112 it would be useful if you could provide a minimal example that demonstrates the problem? The test has not been broken in previous releases so the parallelism maybe is not the only reason.
